radix-transactions/src/model/execution/executable_common.rs

```diff
@@ -110,8 +65,9 @@ pub enum IntentHashNullification {
         expiry_epoch: Epoch,
         ignore_duplicate: bool,
     },
-    /// For where there's no intent hash
-    None,
+    /// For system transactions which currently need to go through
+    /// nullification process.
+    System,
```

Comment:
Very minor - I'm not sure we use this particularly any more - I think you've used `vec![]` in a few places instead - I have a slight preference for there being an ExecutionContextV1 / ExecutionContextV2 in the `new_v1` / `new_v2` constructors, which gets mapped into an ExecutionContext, and e.g. `IntentHashNullificationV1::System` would get mapped to `vec![]`.

If you agree, I can do that in a later PR.

Reply:
This was required for system transactions having to go through the nullifier logic (without actually doing anything). But agreed that this can be done in a better way. Let's defer this and I'll add a TODO to clean this up.